store/postgres/src/deployment_store.rs

```diff
@@ -1880,6 +1855,49 @@ fn resolve_table_name<'a>(layout: &'a Layout, name: &'_ str) -> Result<&'a Table
         })
 }
 
+pub fn generate_index_creation_sql(
+    layout: Arc<Layout>,
+    entity_name: &str,
+    field_names: Vec<String>,
+    index_method: Method,
+    after: Option<BlockNumber>,
+) -> Result<(String, String), StoreError> {
+    let schema_name = layout.site.namespace.clone();
+    let table = resolve_table_name(&layout, &entity_name)?;
+    let (column_names, index_exprs) = resolve_column_names_and_index_exprs(table, &field_names)?;
+
+    let column_names_sep_by_underscores = column_names.join("_");
+    let index_exprs_joined = index_exprs.join(", ");
+    let table_name = &table.name;
+    let index_name = format!(
+        "manual_{table_name}_{column_names_sep_by_underscores}{}",
+        after.map_or_else(String::new, |a| format!("_{}", a))
+    );
+
+    let mut sql = format!(
+        "create index concurrently if not exists {index_name} \
+         on {schema_name}.{table_name} using {index_method} \
+         ({index_exprs_joined}) ",
+    );
+
+    // If 'after' is provided and the table is not immutable, add a WHERE clause for partial indexing
+    if let Some(after) = after {
+        if !table.immutable {
```

Comment:
This would be a tad more readable if it was `if table.immutable` and the branches were swapped. My brain has trouble with double negatives